src/Controller/ResourceController.php

```diff
@@ -160,6 +164,49 @@ public function handle(Request $request) {
           }
         }
 
+        // Process the body for multipart/related request.
+        if ($method == 'put' && !$this->isValidJson($content) && $format != 'stream') {
+          $stream = Stream::factory($request);
+          $parts = MultipartResponseParser::parseMultipartBody($stream);
+          $content = $parts[1]['body'] ?: $content;
+
+          foreach ($parts as $key => $part) {
+            if ($key > 1 && isset($part['headers']['content-disposition'])) {
+              $file_info_found = preg_match('/(?<=\")(.*?)(?=\")/', $part['headers']['content-disposition'], $file_info);
+              if ($file_info_found) {
+                list(, , $file_uuid, $scheme, $filename) = explode('/', $file_info[1]);
+                if ($file_uuid && $scheme && $filename) {
+                  $uri = "$scheme://$filename";
+                  // Check if exists a file with this uuid.
+                  $file = \Drupal::entityManager()->loadEntityByUuid('file', $file_uuid);
+                  if (!$file) {
+                    // Check if exists a file with this $uri, if it exists then
+                    // change the URI and save the new file.
+                    $existing_files = entity_load_multiple_by_properties('file', array('uri' => $uri));
+                    if (count($existing_files)) {
+                      $uri = file_destination($uri, FILE_EXISTS_RENAME);
+                    }
+                  }
+                  if (!$file) {
+                    $file_context = array(
+                      'uri' => $uri,
+                      'uuid' => $file_uuid,
+                      'status' => FILE_STATUS_PERMANENT,
+                      'uid' => \Drupal::currentUser()->id(),
+                    );
+                    $file = $this->serializer()->deserialize($part['body'], '\Drupal\file\FileInterface', 'stream', $file_context);
+                  }
+                  \Drupal::service('plugin.manager.image.effect')->clearCachedDefinitions();
+                  if ($file instanceof FileInterface) {
+                    Cache::invalidateTags(array('file_list'));
+                    $file->save();
```

Comment:
Instead of saving the file directly here in the controller, I think that we should load the attachment resource plugin (`Drupal\relaxed\Plugin\rest\resource\AttachmentResource`) and pass it to the `AttachmentResource::put()` method on that plugin that will be responsible for saving.